pkg/queue/manager.go

```diff
@@ -33,6 +36,8 @@ import (
 	"sigs.k8s.io/kueue/pkg/workload"
 )
 
+const snapshotWorkers = 5
```

Comment:
I would prefer to move the snapshot taking machanism out from manager to clusterqueue_controller. This would mean that we only pass the paramters to clusterqueue_controller. Currently, `queueVisibilityUpdateInterval` is passed to both places. Let me know if there are some complications I'm not aware of.

Reply:
IIUC on both sides we need that parameter anyway to tell us at what interval we need to use snapshot workers (if we want to move them in clusterqueue_controller) and send a new batch of cluster queues.

Also it looks like we need use special struct to put it into snapshot which will contain cq_name and clusterqueue struct itself.

Reply:
I would keep the manager just as a holder for the snapshots. 

So manager would have only: 
- snapshots      map[string][]kueue.ClusterQueuePendingWorkload
- snapshotsMutex sync.RWMutex

Cluster queue manager would have:
- `snapshotsQueue`
- `snapshotWorkers`
- the params: `queueVisibilityUpdateInterval` and `queueVisibilityClusterQueuesMaxCount`
- the Start function

Reply:
moved snapshot logic to cluster queue_controller https://github.com/kubernetes-sigs/kueue/pull/1069/commits/794246738d936bf93cea9a22b48cce7aedb3fdb6